models/eprop_iaf.h

```diff
@@ -0,0 +1,607 @@
+/*
+ *  eprop_iaf.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef EPROP_IAF_H
+#define EPROP_IAF_H
+
+// nestkernel
+#include "connection.h"
+#include "eprop_archiving_node.h"
+#include "eprop_archiving_node_impl.h"
+#include "eprop_synapse.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, e-prop plasticity, current-based, integrate-and-fire
+
+Short description
++++++++++++++++++
+
+Current-based leaky integrate-and-fire neuron model with delta-shaped
+postsynaptic currents for e-prop plasticity
+
+Description
++++++++++++
+
+``eprop_iaf`` is an implementation of a leaky integrate-and-fire
+neuron model with delta-shaped postsynaptic currents used for eligibility
+propagation (e-prop) plasticity.
+
+E-prop plasticity was originally introduced and implemented in TensorFlow in [1]_.
+
+.. note::
+  The neuron dynamics of the ``eprop_iaf`` model (excluding e-prop
+  plasticity) are similar to the neuron dynamics of the ``iaf_psc_delta`` model,
+  with minor differences, such as the propagator of the post-synaptic current
```

Comment:
How come delta PSC needs a propagator?

Reply:
In the original PSC delta model, the propagator is equal to $1$, in the recurrent neuron of the e-prop model there are two propagators for that term (see `P_z_in`) $1$ and $1-\mathrm{exp}(-\mathrm{d}t/\tau_\text{m})$.  There is a PR still under construction to eliminate the second propagator https://github.com/JesusEV/nest-simulator/pull/20.